### PR TITLE
Harden Command Added

### PR DESCRIPTION
**Pull request by afk**
## Description
`harden` command added.
Redirection operator wasn't working, so a handler was written for these operators.

## How Has This Been Tested?
Ran the command on Ubuntu 18.04.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

**Comment by orhun** [comment]
[Interactive Shell <harden()>](https://trello.com/c/Tn6RXgEW/24-interactive-shell-harden)

File: core/cli/commands.py [hunk 1]
```diff
@@ -126,5 +131,17 @@ def do_back(self, arg):
...
+    def do_harden(self, arg):
+        """Execute the hardening method"""
+
+        if not (self.module and self.namespace):
+            logger.error('Select a module/namespace')
+
+        hrd = self.modules[self.namespace][self.module]()
+        cmd = hrd.command()
+
+        logger.info(run_cmd(cmd))
```
Comment (orhun) [inline core/cli/commands.py]: Is it okay for printing executed command's output as `info`? Maybe we can use just `print` since we don't check the output.

**Review by bufgix — APPROVED** [review]

**Review by orhun — APPROVED** [review]

**Review by furk4nyuksel — COMMENTED** [review]
hat
The harden command did not work

**Comment by orhun** [comment]
> hat
> The harden command did not work

It's because you use it in a wrong way.
Correct usage is:
`use disable_rdp`
`harden`

**Comment by furk4nyuksel** [comment]
> > hat
> > The harden command did not work
> 
> It's because you use it in a wrong way.
> Correct usage is:
> `use disable_rdp`
> `harden`

aa

Is working now

**Review by furk4nyuksel — APPROVED** [review]

**Review by orhun — CHANGES_REQUESTED** [review]
I had chance to test this code out and I found a bug.
Code: `self.modules[self.namespace][self.module]()`
What I'm doing is `use logs_restrict_access`
So we can't run harden command unless we select a namespace.
Also it is case sensitive so throws KeyError when you don't care about the capital letters.
Please fix.

**Comment by EnesOkutan** [comment]
Can we solve this problem in a different way?
my suggestion:
By creating a dictionary with name definitions, we can solve it without the need for a loop.
For Example:
dicti = { 'disable_rdp' : 'Disable_RDP' }
Assuming we create such a dict, we can re-create the code as follows:

image

**Comment by EnesOkutan** [comment]
If notative speeds (complexity) are of importance, we can get rid of the loops of these codes in this way.

image

**Comment by orhun** [comment]
> Can we solve this problem in a different way?
> my suggestion:
> By creating a dictionary with name definitions, we can solve it without the need for a loop.
> For Example:
> dicti = { 'disable_rdp' : 'Disable_RDP' }
> Assuming we create such a dict, we can re-create the code as follows:
> 
> image

Which 'problem' are you talking about exactly?

**Comment by EnesOkutan** [comment]
No problem, just an idea of ​​how the code will be healthier.

**Comment by orhun** [comment]
> No problem, just an idea of ​​how the code will be healthier.

Oh, got it.

**Review by orhun — APPROVED** [review]

**Review by EnesOkutan — APPROVED** [review]
Congratulations, it works.

**Pull request merged**